.github/actions/bench/action.yml

```diff
@@ -32,24 +32,47 @@ inputs:
   gh_token:
      description: GitHub access token
      required: true
+  use-nix:
+    description: Whether to run in the default Nix environment
+    default: true
+  custom_shell:
+    description: The shell to use. Only relevant if use-nix is 'false'
+    default: 'bash'
 runs:
   using: composite
   steps:
-      - name: Run benchmark
-        shell: nix develop .#ci -c bash -e {0}
-        run: |
-          tests bench -c ${{ inputs.perf }} --cflags "${{ inputs.cflags }}" --arch-flags "${{ inputs.archflags }}" -v --output output.json ${{ inputs.bench_extra_args }}
-      - name: Dump benchmark
-        shell: bash
-        if: ${{ inputs.store_results != 'true' }}
-        run: |
-          cat output.json
-      - name: Store benchmark result
-        if: ${{ inputs.store_results == 'true' }}
-        uses: benchmark-action/github-action-benchmark@v1
-        with:
-          name: ${{ inputs.name }}
-          tool: 'customSmallerIsBetter'
-          output-file-path: output.json
-          github-token: ${{ inputs.gh_token }}
-          auto-push: true
+    - name: Setup nix
+      if: ${{ inputs.use-nix }}
+      uses: ./.github/actions/setup-nix
+      with:
+        devShell: ci
+        script: |
+          ARCH=$(uname -m)
+          cat >> $GITHUB_STEP_SUMMARY <<-EOF
+            ## Setup
+            Architecture: $ARCH
+            - $(uname -a)
+            - $(nix --version)
+            - $(astyle --version)
```

Comment:
remove this as astyle is not in ci shell

Reply:
done